### DISCUSSION TITLE
Has anyone tried eMBMS end to end with a COTS modem or a phone instead of srsUE?

### DISCUSSION THREAD

**Discussion by nenad38111** [srsran/srsRAN_4G#1327] [General]
First time writing here, perhaps there is a better place for asking for some input/help...so I apologize in advance if this is not correct place.

I do have a COTS phone (and a modem/hotspot) with a special SIM card and it can connect and receive the data from the srsENB (4G)...iperf in TCP mode is working...no issues. VLC streaming in TCP mode also working fine from one point (computer running srsENB/srsEPC/srsmbms) to another point (computer connected to COTS modem runing VLC)

However, I do have the hard time making a phone (or a modem) to receive broadcast from the srsRAN (+blade RF).
The code is the latest as of mid-April 2024.

* There are some discrepancies between the multicast address in the example (eMBMS end-to-end) as it is listed as 239.255.1.1
* The mbms.conf has m1u_multi_addr set to 239.255.0.1. 
* The enb.conf also has default m1u_multiaddr set to 239.255.0.1.

Even when I make this the same ('eMBMS end-to-end' example =>239.255.0.1 -> 239,255.1.1) it is still not working. 

Not working means iperf (in udp modem) does not receive the data on the PC connected to the modem. 
* Wireshark on the transmit side does register 239.255.1.1 packets in UDP on the sgi_mb interface.
* Wireshark on the receive side does not register any packets in UDP
* Comparing to non-broadcast, I do see increased activity in the RF during the broadcast, that means that something 'extra' is really transmitting over the air.

Similar test with VLC ...streaming a video file from computer running srsRAN_4G is not picked up by a phone/modem receiving the stream on the VLC.

Have anyone else tried something like this?
Do I need another setup running srsUE for something like this?

**Comment by caseyonline** [comment]
I just noticed this now after posting my Question about the use of MBMS with a COTS UE,
https://github.com/srsran/srsRAN_4G/discussions/1443
Did you have any luck getting this to work?

**Reply by nenad38111** [reply]
No, I didn't have luck in receiving MBMS traffic on a phone. I have also tried off the shelf 5G modem with more 'knobs' to tweak...but still no luck.
I do have it working on an B210....while 5MHz and higher bandwidths are fine, I have to admit that it has some weird spectrum at 3MHz https://github.com/srsran/srsRAN_4G/issues/1360 and almost certain that this is a bug.

**Reply by caseyonline** [reply]
One of the issues you might have run into is the need for special firmware to enable multicast in Android.   I've been doing some research and it appears that most phones (maybe all) don't ship with that firmware.   Although the chipset supports the MRBs and Android has support you still require a Middleware piece.  I'm looking to track down that piece
